OpenRA.Mods.Cnc/Traits/Minelayer.cs

```diff
@@ -146,11 +146,10 @@ void IResolveOrder.ResolveOrder(Actor self, Order order)
 				minefieldStart = order.ExtraLocation;
 
 				var movement = self.Trait<IPositionable>();
-				var mobile = movement as Mobile;
 
 				var minefield = GetMinefieldCells(minefieldStart, cell, Info.MinefieldDepth)
 					.Where(c => IsCellAcceptable(self, c) && self.Owner.Shroud.IsExplored(c)
-						&& movement.CanEnterCell(c, null, BlockedByActor.Immovable) && (mobile != null && mobile.CanStayInCell(c)))
+						&& movement.CanEnterCell(c, null, BlockedByActor.Immovable) && (movement is Mobile mobile && mobile.CanStayInCell(c)))
```

Comment:
I'm pretty sure the cast was done outside of the Linq clause to avoid having to do it every time the `Where` runs.

Reply:
You're right, I've reverted the change